src/track/globaltrackcache.cpp

```diff
@@ -670,16 +670,11 @@ void GlobalTrackCache::purgeTrackId(TrackId trackId) {
     }
 }
 
-void GlobalTrackCache::evictAndSave(
+void GlobalTrackCache::slotEvictAndSave(
         GlobalTrackCacheEntryPointer cacheEntryPtr) {
+    DEBUG_ASSERT_QOBJECT_THREAD_AFFINITY(this);
     DEBUG_ASSERT(cacheEntryPtr);
 
-    // We need to be sure this is always called from the main thread
-    // because we can only access the DB from it and we must not lose the
-    // the lock until all changes are persistently stored in file and DB
-    // to not hand out the track again with old metadata.
```

Comment:
I think the comment was useful.

Reply:
The comment actually does not belong here. The GlobalTrackCache should not need to care about the actual limitations in other components, mixing up abstraction and isolation.

Instead I have added documentation to `GlobalTrackCacheSaver` that explicitly states the thread affinity aspects and assumptions that matter.

I can try to re-add parts of this comment, but we should not mention any implementation details of the library.

Reply:
See the additional commits. I might also have found and fixed a potential race condition.

Reply:
Ok